### ISSUE TITLE
Failure in Task.all() make recursive calls

### ISSUE THREAD

**Issue by aKentaroIchima** [ReactKit/SwiftTask#68]
When calling Task.all() with many tasks and one of the task was failed, canceling all tasks process will start. But from the current algorithm, it makes recursive call. If the number of tasks is 10000 and the first task is failed, the depth of call is about 10000 and it causes stack overflow. So, the proposal is that prepare flag for limiting the depth of call to be 1.

```
            }.failure { (errorInfo: ErrorInfo) -> Void in

                    lock.lock()
                    _reject(errorInfo)

                    for task in tasks {
                        task.cancel()  // <- this call invokes ".failure" of another task in the same context
                    }
                    lock.unlock()
            }
```

**Comment by inamiy** [comment]
Thanks for reporting!
I just fixed it in #69. Can you take a look?

**Comment by aKentaroIchima** [comment]
It looks good. Thanks!

**Comment by inamiy** [comment]
@aKentaroIchima Merged #69. Thanks for checking 😄
